File: talentlens-backend/src/main/java/com/talentlens/model/ShareRequest.java
package com.talentlens.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;

@Document(collection = "share_requests")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ShareRequest {

    @Id
    private String id;

    private String searchId;
    private String requesterUserId;
    private String ownerUserId;
    private ShareRequestStatus status;
    private Instant requestedAt;
    private Instant resolvedAt;
    private String resolvedBy;
    private String note;
}

File: talentlens-client/src/types/shareRequest.ts
export type ShareRequestStatus = 'PENDING' | 'APPROVED' | 'REJECTED'

export interface ShareRequest {
  id: string
  searchId: string
  searchTitle?: string | null
  requesterUserId: string
  requesterName?: string | null
  ownerUserId: string
  ownerName?: string | null
  status: ShareRequestStatus
  note: string | null
  requestedAt: string
  resolvedAt: string | null
}

export interface CreateShareRequest {
  searchId: string
  recruiterUserId: string
  note?: string
}
